x-pack/platform/plugins/shared/lens/public/editor_frame_service/editor_frame/config_panel/esql_editor.tsx

```diff
@@ -136,6 +137,15 @@ export function ESQLEditor({
     return () => s?.unsubscribe();
   }, [dataLoading$, dispatch, layerId]);
 
+  useEffect(() => {
+    if (isOfAggregateQueryType(query)) {
+      const updatedQuery = fixESQLQueryWithVariables(query.esql, esqlVariables);
```

Comment:
Could it make sense to do this inside the editor component instead of in Lens?

Reply:
I will check it 👍

Reply:
Done here https://github.com/elastic/kibana/pull/213916/commits/6b001b312e758a45158d6c5617607f82aa75ea47